test/functional/ui/searchhl_spec.lua

```diff
@@ -93,6 +93,59 @@ describe('search highlighting', function()
     ]])
   end)
 
+  it('highlights after EOL', function()
+    insert("\n\n\n\n\n\n")
+
+    feed("gg/^<cr>")
+    screen:expect([[
+      {2: }                                       |
+      {2:^ }                                       |
+      {2: }                                       |
+      {2: }                                       |
+      {2: }                                       |
+      {2: }                                       |
+      /^                                      |
+    ]])
+
+    -- Test that highlights are preserved after moving the cursor.
+    feed("j")
+    screen:expect([[
+      {2: }                                       |
+      {2: }                                       |
+      {2:^ }                                       |
+      {2: }                                       |
+      {2: }                                       |
+      {2: }                                       |
+      /^                                      |
+    ]])
+
+    -- Repeat the test in rightleft mode.
+    feed_command("nohlsearch")
+    feed_command("set rightleft")
+    feed("gg/^<cr>")
+
+    screen:expect({
```

Comment:
is there any reason the ordinary string form can't be used? Does `screen:snapshot_util()` generate something that is wrong?

Reply:
Oops, I think an earlier iteration of this test might have had all lines starting with whitespace, but you're right - this version works fine with the ordinary string form. Reverted the extra test changes.